### PR TITLE
Task 4 testing

### PR DESCRIPTION
**Pull request by sixtyXi**

### CODE REVIEW

File: src/list-movie/components/SortContainer/SortContainer.jsx [hunk 1]
```diff
@@ -5,26 +5,35 @@ import { FilterPanel } from '../../../components/FilterPanel/FilterPanel';
...
   render() {
+    const { filterValue } = this.state;
+
     return (
       <FilterPanel
-        name="sortValue"
-        handleChange={this.handleChange.bind(this)}
-        title="Sort by"
+        name="filterValue"
+        handleChange={e => this.handleChange(e)}
```
Comment (IrinaShilina) [inline src/list-movie/components/SortContainer/SortContainer.jsx]: Avoid using arrow functions in the render method. Just `handleChange={this.handleChange}` should be enough.

File: src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const { searchTxt } = this.state;
+
+    return (
+      <div className={styles.searchField}>
+        <input
+          data-cy="searchField"
+          type="text"
+          value={searchTxt}
+          id="search"
+          onChange={e => this.handleChange(e)}
```
Comment (IrinaShilina) [inline src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx]: Please avoid arrow functions in the render method.

File: src/list-movie/components/SearchFieldContainer/SearchFieldContainer.test.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+describe('SearchFieldContainer', () => {
+  test('renders correctly', () => {
+    const wrapper = shallow(<SearchFieldContainer />);
+
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  test('calls *handleChange* on input value changes', () => {
+    const wrapper = mount(<SearchFieldContainer />);
```
Comment (IrinaShilina) [inline src/list-movie/components/SearchFieldContainer/SearchFieldContainer.test.jsx]: Do you really need 'mount' here? Is it now working with:
```
const wrapper = shallow(<SearchFieldContainer />);
const instance = wrapper.instance();
```

Also, maybe you can only shallow render wrapper once, before the first 'test' block. It's not an error, but may save time for a test run:
```
describe('SearchFieldContainer', () => {
 const wrapper = shallow(<SearchFieldContainer />);
 const instance = wrapper.instance();

 test('renders correctly', () => {
```

File: src/list-movie/components/SortContainer/SortContainer.jsx [hunk 4]
```diff
@@ -5,26 +5,35 @@ import { FilterPanel } from '../../../components/FilterPanel/FilterPanel';
...
     super(props);
-    this.filters = [ 'release date', 'rating' ];
-    this.defaultFilter = this.filters[0];
+    this.filters = props.filters;
+
+    if (!this.filters.length) return;
+
+    this.defaultFilterIdx = props.defaultFilterIdx || 0;
+    this.panelTitle = props.title || 'Search by';
+    this.panelType = props.type || 'btn';
```
Comment (IrinaShilina) [inline src/list-movie/components/SortContainer/SortContainer.jsx]: You can replace this with default props (see React's PropTypes)
https://reactjs.org/docs/typechecking-with-proptypes.html

File: src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+
+import styles from './SearchFieldContainer.scss';
+
+export class SearchFieldContainer extends Component {
+  state = {
+    searchTxt: ''
+  };
+
+  handleChange(event) {
```
Comment (IrinaShilina) [inline src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx]: Arrow function here:
```
handleChange = event => this.setState({ searchTxt: event.target.value });
```

File: src/list-movie/components/SortContainer/SortContainer.jsx [hunk 6]
```diff
@@ -5,26 +5,35 @@ import { FilterPanel } from '../../../components/FilterPanel/FilterPanel';
...
+
     return (
       <FilterPanel
-        name="sortValue"
-        handleChange={this.handleChange.bind(this)}
-        title="Sort by"
+        name="filterValue"
+        handleChange={e => this.handleChange(e)}
+        title={this.panelTitle}
         filters={this.filters}
```
Comment (IrinaShilina) [inline src/list-movie/components/SortContainer/SortContainer.jsx]: Why not `this.props.filters`?

Comment (sixtyXi) [reply src/list-movie/components/SortContainer/SortContainer.jsx]: ESLint preferences restrict using props without destructuring )

File: src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx (hunk 2 above)
Comment (sixtyXi) [reply src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx]: Done.

File: src/list-movie/components/SearchFieldContainer/SearchFieldContainer.test.jsx (hunk 3 above)
Comment (sixtyXi) [reply src/list-movie/components/SearchFieldContainer/SearchFieldContainer.test.jsx]: Thanks! I moved wrapper rendering to beforeEach().

File: src/list-movie/components/SortContainer/SortContainer.jsx (hunk 4 above)
Comment (sixtyXi) [reply src/list-movie/components/SortContainer/SortContainer.jsx]: I am going to install and use this library in the next task.

File: src/list-movie/components/SortContainer/SortContainer.jsx (hunk 1 above)
Comment (sixtyXi) [reply src/list-movie/components/SortContainer/SortContainer.jsx]: Done.

File: src/list-movie/components/SortContainer/SortContainer.jsx (hunk 6 above)
Comment (IrinaShilina) [reply src/list-movie/components/SortContainer/SortContainer.jsx]: Then try
```
const { filterValue } = this.state;
const { filters } = this.props;
```

It's not a big deal, though. You can fix it in the next task.

**Pull request merged**